app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -537,8 +537,8 @@ BrowserWorld::State::UpdateControllers(bool& aRelayoutWidgets) {
       controller.modelToggle->ToggleAll(controller.mode == ControllerMode::Device);
 
     if (handMeshRenderer) {
-      handMeshRenderer->Update(controller, controllers->GetRoot(),
-                               controller.enabled && controller.mode == ControllerMode::Hand);
+      handMeshRenderer->Update(controller.index, controller.handJointTransforms, controllers->GetRoot(),
+                               controller.enabled && controller.mode == ControllerMode::Hand, controller.leftHanded);
```

Comment:
The idea here is not leaking the `Controller` object to the backend. I don't like this change much either, but I suppose there is a reason we pass a delegate (`DeviceDelegate`) to the backend to mediate and fill-up `Controller` data. 

I'm fine passing the Controller object directly, but it should be noted that it would be the first time we do that.